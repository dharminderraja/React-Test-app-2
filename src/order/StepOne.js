import React, { Component } from 'react';

const MAX = 10;

class StepOne extends Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      meal: '',
      number: 1,
      errors: {}
    };
  }

  componentWillMount() {
    const { number, meal } = this.props.data || {};
    if (number && meal) {
      this.setState({ number, meal });
    }
  }

  onChange(property, value) {
    this.setState({ [property]: value });
  }

  validate() {
    let { errors, meal, number } = this.state;

    errors['meal'] = !meal ? 'Please select meal' : '';
    errors['number'] = number > 10 || number < 1 ? 'Please input number in range from 1 to 10' : '';
    this.setState({ errors })

    return Object.keys(errors).every(k => !errors[k]);
  }

  nextStep() {
    const { updateData, nextTab, disableTabs } = this.props;
    const { number, meal } = this.state;

    if (this.validate()) {
      updateData({ number, meal });
      nextTab();
    } else {
      disableTabs(['tabTwo', 'tabThree', 'tabFour']);
    }
  }

  render() {
    const { meals } = this.props;
    const { number, meal, errors } = this.state;

    return (
      <div className='step-detail'>
        <div>
          <span className='title'>Please select a meal</span>
          <select className='meal' value={meal || ''} onChange={evt => this.onChange('meal', evt.target.value)}>
            <option value=''>...</option>
            {(meals || []).map((i, idx) => (<option key={idx} value={i}>{i}</option>))}
          </select>
          {errors['meal'] && <span className='alert-warning'>{errors['meal']}</span>}
        </div>
        <div>
          <span className='title'>Please Enter Number of people</span>
          <input className='number-meal' value={number} onChange={evt => this.onChange('number', evt.target.value)} type='number' min='1' max={MAX} />
          {errors['number'] && <span className='alert-warning'>{errors['number']}</span>}
        </div>
        <div className='nav-step'>
          <button className='next' onClick={() => this.nextStep()}>Next</button>
        </div>
      </div>
    );
  }
}

export default StepOne;
