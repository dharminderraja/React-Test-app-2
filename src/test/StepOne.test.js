import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import StepOne from '../order/StepOne';

describe('Step One', () => {
  it('should be defined', () => {
    expect(StepOne).toBeDefined();
  });

  it('should render correctly', () => {
    const component = renderer.create(<StepOne />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show values correctly', () => {
    const tree = shallow(
      <StepOne
        meals={['breakfast', 'lunch', 'dinner']}
        data={{ meal: 'lunch', number: 2 }} />
    );

    expect(tree.find('.step-detail .meal').children()).toHaveLength(4);
    expect(tree.find('.step-detail .meal').getElement().props.value).toEqual('lunch');
    expect(tree.find('.step-detail .number-meal').getElement().props.value).toEqual(2);
  });
});