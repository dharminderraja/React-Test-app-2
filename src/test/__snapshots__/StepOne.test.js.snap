// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step One should render correctly 1`] = `
<div
  className="step-detail"
>
  <div>
    <span
      className="title"
    >
      Please select a meal
    </span>
    <select
      className="meal"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        ...
      </option>
    </select>
  </div>
  <div>
    <span
      className="title"
    >
      Please Enter Number of people
    </span>
    <input
      className="number-meal"
      max={10}
      min="1"
      onChange={[Function]}
      type="number"
      value={1}
    />
  </div>
  <div
    className="nav-step"
  >
    <button
      className="next"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
